lsp-modeline.el

```diff
@@ -61,49 +70,60 @@
-                                                    lsp-modeline--code-action->string))
-                                           'face 'lsp-modeline-code-actions-face))
-          (single-action? (= (length actions) 1))
+  (-let* ((single-action? (= (length actions) 1))
           (keybinding (concat "("
                               (-some->> #'lsp-execute-code-action
                                 where-is-internal
                                 (-find (lambda (o)
                                          (not (member (aref o 0) '(menu-bar normal-state)))))
                                 key-description)
                               ")"))
-          (string (if single-action?
-                      (concat " " icon " " first-action-string " ")
-                    (concat " " icon " " first-action-string " "
-                            (propertize (concat "(" (number-to-string (1- (seq-length actions))) " more)")
-                                        'display `((height 0.9))
-                                        'face 'lsp-modeline-code-actions-face)
-                            " "))))
-    (propertize string
-                'help-echo (concat (format "Apply code actions %s\nmouse-1: " keybinding)
-                                   (if single-action?
-                                       first-action-string
-                                     "select from multiple code actions"))
-                'mouse-face 'mode-line-highlight
-                'local-map (make-mode-line-mouse-map
-                            'mouse-1 (lambda ()
-                                       (interactive)
+          (built-string (lsp-modeline--build-code-actions-segments actions)))
+    (add-text-properties 0 (length built-string)
+                         (list 'help-echo
+                               (concat (format "Apply code actions %s\nmouse-1: " keybinding)
                                        (if single-action?
-                                           (lsp-execute-code-action (lsp-seq-first actions))
-                                         (lsp-execute-code-action (lsp--select-action actions))))))))
+                                           (lsp-modeline--preferred-code-action-name actions)
+                                         "select from multiple code actions"))
+                               'mouse-face 'mode-line-highlight
+                               'local-map (make-mode-line-mouse-map
+                                           'mouse-1 (lambda ()
+                                                      (interactive)
+                                                      (if single-action?
+                                                          (lsp-execute-code-action (lsp-seq-first actions))
+                                                        (lsp-execute-code-action (lsp--select-action actions))))))
+                         built-string)
+    (unless (string= "" built-string)
```

Comment:
It seems that the prefix space is not prepend? Or did I miss something?

Reply:
Yeah, it was missing, done @kiennq !